proposals/collection-literals.md

```diff
@@ -297,14 +337,17 @@ var y = First(new int[0], [1, 2, 3]);    // ok: int[]
 static T First<T>(T x, T y) => x;
 ```
 
-To allow conversions to other collection types, type inference may need to infer exact, lower-bound, and upper-bound inferences from the **expression `U`** to a type `V` when an expression is given, and *fixing* §11.6.3.12 updates the sets of *candidate types* based on implicit conversion from the **expression** bound to the each *candidate type*.
+To allow conversions to other collection types, type inference may need to infer exact, lower-bound, and upper-bound inferences from the **expression `U`** (rather than the type of `U`) to a type `V` when an expression is given, and *fixing* §11.6.3.12 updates the sets of *candidate types* based on implicit conversion from the **expression** bound to the each *candidate type*.
 
 _Should nested cases infer collection types successfully?_
 ```c#
 var x = new[] { [ulong.MaxValue], [1, 2, 3] };  // ok: List<ulong>[]
 var y = First([[ulong.MaxValue]], [[1, 2, 3]]); // ok: List<List<ulong>>
 ```
 
+## Overload resolution
+_Include betterness order - perhaps the following, from best to worst: spans; arrays and constructed types; interface types._
```

Comment:
sounds good.  i think one good thing to think-about/design/test is:

```c#
void Foo(IEnumerable<KVP<X,Y>> values);
void Foo(IDictionary<X,Y> values);
```

if we do `Foo([1:2])`, which does that pick?  Waht should it pick?  Should it error?

Reply:
Added example, thanks.

Reply:
[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQAYACdbAFgG4CD0BmMzEgYRIG8CTuyKSBZIgAoAlBwFCYAUwDuJACJQAxgBcoAexgQATgE8APOSJwSsFQD5RIqiQC+XHg+4NeE8vX0BpKboBqEABsAVykABWhtQ2xjUxgLcxIABwiAZzFOfB47JzJGdD5+IQBJRVUNLT0omLME5KhtNI4c+3xbIA)

I haven't reviewed the overload resolution spec to confirm, but I think more-derived interfaces are *better*. It feels reasonable to apply the same rule here, even if different construction strategies would potentially be used for each target type.